pkg/controller/direct/logging/logmetric_controller.go

```diff
@@ -173,7 +173,8 @@ func (a *logMetricAdapter) Create(ctx context.Context, u *unstructured.Unstructu
 		}
 	}
 
-	logMetric := convertKCCtoAPI(a.desired)
+	logMetric := convertKCCtoAPI(&a.desired.Spec)
+	logMetric.Name = a.resourceID
```

Comment:
I like this. Shall this be our general solution for resourceID <-> resource name mapping? 
Also, I'm wondering what we have and have not determined in the previous discussions https://github.com/GoogleCloudPlatform/k8s-config-connector/pull/1721#discussion_r1594810095